ui/cypress/support/pages/pageFacory.ts

```
@@ -0,0 +1,20 @@
+import { CATALOG_TYPES, PAGES } from "../types/types";
+import { CatalogPage } from "./catalogPage";
+import { GpuPage } from "./gpuPage";
+import { LaptopPage } from "./laptopPage";
+
+
+export class PageFactory {
+    static getPage(pageName: PAGES){
+        switch (pageName) {
+            case PAGES.CATALOG_TYPES:
+                return new CatalogPage()
+            case PAGES.GPU:
+                return new GpuPage()    
+            case PAGES.LAPTOP:
+                return new LaptopPage()    
+            default:
+                break;

```

Please throw an error here saying that invalid page name is provided.